code/src/components/HappyForm.js

```diff
@@ -0,0 +1,47 @@
+import React, { useState } from 'react'
+
+import { CharacterCount } from "./CharacterCount"
+import "./happyForm.css"
+
+const url = "https://happy-thoughts-technigo.herokuapp.com/thoughts"
+
+export const HappyForm = (props) => {
+    const [message, setMessage] = useState("")
+
+    const handleSubmit = event => {
+        event.preventDefault()
+        fetch(url, {
+            method: "POST",
+            body: JSON.stringify({ message }),
+            headers: { "Content-Type": "application/json" }
+        })
+            .then(() => {
+                setMessage("")
+                props.onFormSubmit(message)
+            })
+            .catch(err => console.log("error:", err))
+    }
+
+    const emptyMessage = (value) => value.replace(/\s/g, "").length === 0;
```

Comment:
`isMessageEmpty` would be better from the semantic point of view of how we name boolean functions